### ISSUE TITLE
[QE] Reportportal launch for failed pipeline run

### ISSUE THREAD

**Issue by lilyLuLiu** [crc-org/crc#4412]
Currently, if the pipeline run failed before `reportportal-import` task, there is no record of the failure(as pipeline run periodically clean).
Add a post task for pipeline run if it's failed. This task will create a XML file that include the failure details and import the xml file to reportportal.

**Comment by albfan** [comment]
Do we have any example of this error?

**Comment by lilyLuLiu** [comment]
We aim to using reportportal to check all tests as ocp clear pipeline runs in one or two days. But if pipeline failed before task `reportportal-import`(like the picture), there is nothing imported into reportportal. We even don't know there is a test run failed from reportportal. 
![Image](https://github.com/user-attachments/assets/89c09b99-4b0e-4aca-8075-61064a4ece92)

So we need a [final task](https://tekton.dev/docs/pipelines/pipelines/#adding-finally-to-the-pipeline:~:text=the%20Pipeline.-,Adding%20Finally%20to%20the%20Pipeline,-You%20can%20specify) that triggered if the pipeline run failed. This task will collect the failed info and import to reportportal.

**Comment by lilyLuLiu** [comment]
* To gather the failed pipeline-run logs, the task needs in the final block. 
* To upload the pipeline-run logs to reportportal, the task should run before the `reporportal` task.
* Also, removing the folder in PVC should be executed after the `reportportal` task
* But in the final block, all tasks run in parallel.  Can't realize get pipeline-run log first, then upload to reportportal  

So I plan to upgrade the `reportportal` task, it will contain 4 steps in the task
1. get current pipeline logs (disabled in default)
2. transfer the log to XML file (disabled in default)
3. upload all XML files to reporportal 
4. remove the pvc folder (disabled in default)

@adrianriobo WDYT?

**Comment by adrianriobo** [comment]
Yeah so I would suggest a different cause one key issue right now is that when you provision a machine the state used by mapt on our pipelines is stored on the same PVC..so if you delete the content before running the decommission we may end up on situation where machines from mapt will not be detroyed.

Also this behavior of using the workspace (PVC) to keep the state and also move the credentials can be changed and actually we should change it (to be in line with best practices). One is to keep the state on s3 and the other is move the info required to connect to the machine from workspace to create secrets holding it. This is ready but not merged here https://github.com/redhat-developer/mapt/pull/336

So with all said I would suggest:

Prioritize to move https://github.com/redhat-developer/mapt/pull/336 in as so we will remove the PVC dependency for the state (this will require touch on each pipeline we are currently using)

Then you can create simple cleanup task which can be executed in parallel on the final block after the sync of the information has been completed on repor portal and s3

WDYT?

**Comment by albfan** [comment]
Cannot find remove-pvc-resource source on crc-qe-virtualized

Looks it is applied on openshift cluster and conflicts with decommision task

https://console-openshift-console.apps.gpc.ocp-hub.prod.psi.redhat.com/k8s/ns/devtoolsqe--pipeline/tekton.dev~v1~PipelineRun/release-2.47.0-rhel93-f7c5xk

causing machines to not be decommisioned.

can we add a condition to be disabled by default?

At this moment this only affects to release test, so nightly is not affected

**Comment by lilyLuLiu** [comment]
@albfan `remove-pvc-resource` is created in https://gitlab.cee.redhat.com/crc/qe-platform/-/merge_requests/274
I have reverted the change in cluster.

**Comment by lilyLuLiu** [comment]
Upload MR that create a reportportal v1.2.0
https://gitlab.cee.redhat.com/crc/qe-platform/-/merge_requests/279
